### PR TITLE
Refactor CI to support Python 3.10

### PR DESCRIPTION
**Pull request by arcondello**

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,3 +1,5 @@
-numpy==1.19.4; python_version >= '3.6'
-cython==0.29.21
-dimod==0.9.11
+numpy==1.19.1;python_version<="3.9"
+numpy==1.21.4;python_version>="3.10"
```
Comment (randomir) [inline requirements.txt:2]: There's a gap `<3.9, 3.10>`.

Comment (arcondello) [reply requirements.txt:2]: Seems to work fine? For instance the CI 3.9.4 job correctly installs NumPy 1.19.1

Comment (arcondello) [reply requirements.txt:2]: Though I should actually change it to 3.8. Can use the other syntax if you feel like it's clearer.

Comment (randomir) [reply requirements.txt]: If that works, I need to revisit my version specifiers :scream: 

But yes, I think this is much clearer:
```suggestion
numpy==1.19.1; python_version < "3.10"
numpy==1.21.4; python_version >= "3.10"
```

Comment (arcondello) [reply requirements.txt]: It does indeed work. But anyway, have already made the change.

Comment (randomir) [reply requirements.txt]: Oh, this is [such a fail](https://www.python.org/dev/peps/pep-0496/#version-numbers):
- `python_version` is `platform.python_version()[:3]`
- only `python_full_version` includes the patch version

Comment (randomir) [reply requirements.txt]: Not to mention that `[:3]` translates `3.10.1` to `3.1` :laughing:

**Pull request merged**